errors.go

```diff
@@ -11,41 +11,46 @@ import (
 type (
 	// Errors may be generated during deserialization, binding,
 	// or validation. It implements the built-in error interface.
-	Errors []Error
+	Errors []FieldError
 
 	// Error is a powerful implementation of the built-in error
 	// interface that allows for error classification, custom error
 	// messages associated with specific fields, or with no
 	// associations at all.
-	Error struct {
-		// An error supports zero or more field names, because an
-		// error can morph three ways: (1) it can indicate something
-		// wrong with the request as a whole, (2) it can point to a
-		// specific problem with a particular input field, or (3) it
-		// can span multiple related input fields.
-		FieldNames []string `json:"fieldNames,omitempty"`
+	fieldError struct {
+		// An error supports a field names because an
+		// error can morph two ways: (1) it can indicate something
+		// wrong with the request as a whole, or (2) it can point to a
+		// specific problem with a particular input field.
+		fieldName string `json:"fieldNames,omitempty"`
```

Comment:
Yes, I can put that back. The reason I took it out was that there was nowhere in the package that set the `FieldNames` to a slice with more than one element, and any consumer of the package could easily group the field errors by classification to aggregate them if they wanted. But it's not a change I feel strongly about, and is arguably out of scope here.